openpype/hosts/substancepainter/api/lib.py

```diff
@@ -643,3 +643,55 @@ def _setup_prompt():
         return
 
     return project_mesh
+
+
+def has_rgb_channel_in_texture_set(texture_set_name, map_identifier):
+    """Function to check whether the texture has RGB channel.
+
+    Args:
+        texture_set_name (str): Name of Texture Set
+        map_identifier (str): Map identifier
+
+    Returns:
+        colorspace_dict: A dictionary which stores the boolean
+            value of textures having RGB channels
+    """
+    texture_stack = (
+        substance_painter.textureset.Stack.from_name(texture_set_name)
+    )
+    # 2D_View is always True as it exports all texture maps
+    colorspace_dict = {"2D_View": True}
+    colorspace_dict["BaseColor"] = texture_stack.get_channel(
+        substance_painter.textureset.ChannelType.BaseColor).is_color()
+    colorspace_dict["Roughness"] = texture_stack.get_channel(
+        substance_painter.textureset.ChannelType.Roughness).is_color()
+    colorspace_dict["Metallic"] = texture_stack.get_channel(
+        substance_painter.textureset.ChannelType.Metallic).is_color()
+    colorspace_dict["Height"] = texture_stack.get_channel(
+        substance_painter.textureset.ChannelType.Height).is_color()
+    colorspace_dict["Normal"] = texture_stack.get_channel(
+        substance_painter.textureset.ChannelType.Normal).is_color()
+    return colorspace_dict.get(map_identifier, False)
+
+
+def texture_set_filtering(texture_set_same, template):
+    """Function to check whether some specific textures(e.g. Emissive)
+    are parts of the texture stack in Substance Painter
+
+    Args:
+        texture_set_same (str): Name of Texture Set
+        template (str): texture template name
+
+    Returns:
+        texture_filter: A dictionary which stores the boolean
+            value of whether the texture exist in the channel.
+    """
+    texture_filter = {}
+    channel_stack = substance_painter.textureset.Stack.from_name(
+        texture_set_same)
+    has_emissive = channel_stack.has_channel(
+        substance_painter.textureset.ChannelType.Emissive)
+    map_identifier = strip_template(template)
+    if map_identifier == "Emissive":
+        texture_filter[map_identifier] = has_emissive
+    return texture_filter.get(map_identifier, True)
```

Comment:
```suggestion
def texture_set_filtering(texture_set_name, template):
    """Function to check whether some specific textures(e.g. Emissive)
    are parts of the texture stack in Substance Painter

    Args:
        texture_set_name (str): Name of Texture Set
        template (str): texture template name

    Returns:
        texture_filter: A dictionary which stores the boolean
            value of whether the texture exist in the channel.
    """
    texture_filter = {}
    channel_stack = substance_painter.textureset.Stack.from_name(
        texture_set_name)
    has_emissive = channel_stack.has_channel(
        substance_painter.textureset.ChannelType.Emissive)
    map_identifier = strip_template(template)
    if map_identifier == "Emissive":
        texture_filter[map_identifier] = has_emissive
    return texture_filter.get(map_identifier, True)
```

It's good to be aware that **strip_template** isn't a _very reliable way! to get the exact internal channel type name in Substance Painter. It will parse the output filename of the output template - it does not in any way refer to an internal name of Substance Painter's API. As such, I'd say this function implementation - which relies on the `map_identifier` being resolved like that is too unreliable to merge.

It will not work if the output template writes it as `emmision` or `EM` or whatever other terminology one might use for this in their templates. For example these wouldn't work:
```
"$textureSet_e(.$udim)"
"$textureSet_em(.$udim)"
"$textureSet_emission(.$udim)"
"$textureSet_emissionMap(.$udim)"
"$textureSet_light(.$udim)"
"$textureSet_glow(.$udim)"
```
Or whatever output template someone would use for their file naming.

This is unfortunately due how the substance painter API does not allow us to query what particular type a particular channel is in the output templates (at least I hadn't found that - if there IS, then that's gold because we should refactor other bits as well!)

Reply:
> This is unfortunately due how the substance painter API does not allow us to query what particular type a particular channel is in the output templates (at least I hadn't found that - if there IS, then that's gold because we should refactor other bits as well!)

maybe we can add some TODO comment to remind us to fix it later. I dont like the way it is being hardcoded.

Reply:
I would just **not** merge this - it's so specific to _one_ case that it doesn't make much sense. Rather fix it so it solves it for any channel if this comes up, etc.